### PR TITLE
Kp/multilabel

### PR DESCRIPTION
**Pull request by kprokofi**
WIP
- [x] Updated handling the optimization parameters by optimizer + NBD
- [x] Load model`s weights directly from the link in the config
- [x] fast_ai lr_finder fix 
- [x] A new early stopping for multilabel task

### CODE REVIEW

**Comment by kprokofi** [comment]
@sovrasov , ready for review

File: torchreid/utils/torchtools.py [hunk 1]
```diff
@@ -307,74 +437,6 @@ def _print_loading_weights_inconsistencies(discarded_layers, unmatched_layers):
         )
 
 
-def load_pretrained_weights(model, file_path='', pretrained_dict=None):
```
Comment (sovrasov) [inline torchreid/utils/torchtools.py:310]: Let's avoid changing the functions declaration order to produce a compact git diff

File: torchreid/models/timm_wrapper.py [hunk 2]
```diff
@@ -75,6 +76,20 @@ def infer_head(self, x):
         self.dropout(x)
         return self.classifier(x.view(x.shape[0], -1))
 
+    def get_config_optim(self, lrs):
+        parameters = [
+            {'params': self.model.named_parameters()},
+        ]
+        if isinstance(lrs, list):
+            assert len(lrs) == len(parameters)
+            for lr, param_dict in zip(lrs, parameters):
```
Comment (sovrasov) [inline torchreid/models/timm_wrapper.py:85]: len(parameters) is always 1, so why the default implementation from the parent class won't work and we need a replacement?

File: torchreid/optim/lr_finder.py [hunk 3]
```diff
@@ -119,7 +119,7 @@ def fast_ai(self):
         if self.epochs_warmup != 0:
             get_model_attr(self.model, 'to')(self.model_device)
             print("Warmup the model's weights for {} epochs".format(self.epochs_warmup))
-            self.engine.run(max_epoch=self.epochs_warmup, lr_finder=self.engine_cfg, stop_callback=self.stop_callback)
+            self.engine.run(max_epoch=self.epochs_warmup, lr_finder=self.engine_cfg, stop_callback=self.stop_callback,eval_freq=1)
```
Comment (sovrasov) [inline torchreid/optim/lr_finder.py]: A space is needed before the last arg

File: torchreid/optim/optimizer.py [hunk 4]
```diff
@@ -131,36 +132,59 @@ def _build_optim(model,
...
+            param_group['weight_decay'] = weight_decay
+            layer_lr = param_group['lr']
+            layer_params = param_group['params']
+            for name, param in layer_params:
+                if param in params_set:
+                    raise RuntimeError(f"the parameter {name} duplicated")
+                else:
+                    params_set.add(param)
+
+        param_groups.append({'params': list(params_set), 'lr': layer_lr, 'weight_decay': weight_decay})
```
Comment (sovrasov) [inline torchreid/optim/optimizer.py]: We shouldn't rely on internal variable from the loop body, I suggest using lr instead of layer_lr.

Comment (kprokofi) [reply torchreid/optim/optimizer.py]: I've mistaken it with an indent. This `append` is supposed to be inside the loop

Comment (sovrasov) [reply torchreid/optim/optimizer.py]: Now `params_set` contains params from the groups processed on previous iterations, so some parameters could be in several group at the same time

**Review by sovrasov — APPROVED** [review]

**Pull request merged**